### DISCUSSION TITLE
What should the header extension of choice for C++2 files be?

### DISCUSSION THREAD

**Discussion by isidorostsa** [hsutter/cppfront#1091] [Q&A] [answered]
In C++ we usually use `.hpp` however many C++ projects do use `.h` files. What should we use for C++2? 

Posting here to gauge the consensus, and to discuss about what files the C++2 syntax highlighters should color. 

Relevant issue in the C++ syntax highlighter extension for VSCode: https://github.com/elazarcoh/cpp2-syntax/pull/6

**Comment by vovaskochko** [comment]
Currently we have .h2

**Reply by isidorostsa** [reply]
That has been my impression from looking at the few cpp2 projects that exist.

**Reply by hsutter** [reply, marked as answer]
Yes, `.h2` is what I've been using. It would be easy to extend to support `.hpp2` and `.hxx2`, but I'd like to wait to see if there's real demand (e.g., that maybe it breaks some tools that only work with `.hpp` or `.hxx`?) before complicating it.

**Comment by SebastianTroy** [comment]
As cpp2 is targeting module based compilation, aren't header files just a tempory workaround until compilers support modules properly?


On 30 May 2024 22:11:17 Isidoros ***@***.***> wrote:

That has been my impression from looking at the few cpp2 projects that exist.

**Reply by hsutter** [reply]
> As cpp2 is targeting module based compilation, aren't header files just a tempory workaround until compilers support modules properly?

Cpp2 aims to be a "modules first" design, yes. One constraint is that modules aren't quite ready for prime time yet, which is why I haven't yet implemented module/export-related grammar in Cpp2. Also, even when they arrive and mature they'll still be "in addition to header files"... header files will be actively used and depended on by some libraries and code bases for decades to come and so will need to be supported long-term even just as a matter of compatibility.
